### PR TITLE
Add Codecov functionality to master

### PR DESCRIPTION
**Pull request by imscaradh**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,34 +104,6 @@
                 </plugin>
                 <!--This plugin's configuration is used to store Eclipse m2e settings 
 					only. It has no influence on the Maven build itself. -->
-                <plugin>
```
Comment (Tschanzt) [inline pom.xml]: @imscaradh  Why was this removed? Can you make a seperate commit out of this?

**Comment by Tschanzt** [comment]
@imscaradh could you squash the commits? It looks a bit messy right now.

File: pom.xml (hunk 1 above)
Comment (imscaradh) [inline pom.xml]: Because nobody of us uses Eclipse, therefore we don't need that plugin.
